bin/check-solr-replication-lag.rb

```diff
@@ -55,7 +75,8 @@ def get_url_json(url, notfoundok)
   end
 
   def run
-    uri = "#{config[:core_url]}/replication?command=details&wt=json"
+    base_core_uri = (config[:secure] ? 'https' : 'http') + "://#{config[:host]}:#{config[:port]}/solr/#{config[:core]}"
```

Comment:
why do we need a conditional? Just pass what they define, if you are worried about someone passing in an invalid protocol you can control the acceptable values as I show above.